### PR TITLE
Prepare switching to immutable `ConfigMap`s/`Secret`s

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area robustness
/kind enhancement

**What this PR does / why we need it**:
This PR
* activates the new garbage collector controller in the `gardener-resource-manager` (https://github.com/gardener/gardener-resource-manager/pull/127) with a sync period of `12h`
* introduces a utility function for computing the checksum of the `data` in `ConfigMap`s (earlier, we only had such function for `Secret`s)
* introduces a utility function for making a `ConfigMap` or `Secret` immutable

**Which issue(s) this PR fixes**:
Part of #3484

**Special notes for your reviewer**:
✅ Depending on a new release (+ revendoring) of `gardener/gardener-resource-manager` (see #4373)

/squash

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Review by timebertt — COMMENTED** [review]
Looks good already :)

File: pkg/utils/kubernetes/object.go [hunk 1]
```diff
@@ -109,3 +115,37 @@ func CreateResetObjectFunc(obj runtime.Object, scheme *runtime.Scheme) (func(),
...
+			return "-"
+		}
+		labels = map[string]string{references.LabelKeyGarbageCollectable: references.LabelValueGarbageCollectable}
+	)
+
+	switch o := obj.(type) {
+	case *corev1.Secret:
+		o.Immutable = pointer.Bool(true)
+		o.Name += prependHyphen(o.Name) + utils.ComputeSecretChecksum(o.Data)[:numberOfChecksumChars]
+		o.Labels = utils.MergeStringMaps(o.Labels, labels)
```
Comment (timebertt) [inline pkg/utils/kubernetes/object.go]: Why not use `metav1.SetMetaDataLabel(&o.ObjectMeta, references.LabelKeyGarbageCollectable, references.LabelValueGarbageCollectable)`? No need to merge here, right?

**Review by timebertt — COMMENTED** [review]
One nit. Rest lgtm

File: pkg/utils/kubernetes/object.go (hunk 1 above)
Comment (rfranzke) [reply pkg/utils/kubernetes/object.go]: 😄 Right, thanks, done.

**Review by timebertt — APPROVED** [review]
/lgtm

**Pull request merged**